### PR TITLE
Review Article by Benjamminj: JavaScript Map

### PR DESCRIPTION
**Pull request by rwieruch**
Hey @benjamminj I will add a couple of commits here to refine the article if that's okay with you. I will let you know when I am done so that you can check it again yourself :)

### CODE REVIEW

**Comment by benjamminj** [comment]
Awesome, sounds great! Thanks a bunch @rwieruch 👏

File: javascript-map.md [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+
+const newArray = originalArray.map(function (number) {
+  return number + 1;
+});
+
+console.log(originalArray); // [1, 2, 3, 4, 5]
+console.log(newArray); // [2, 3, 4, 5, 6]
+{{< /highlight >}}
+
+However, if you decide to extract the callback function as standalone fucntion declared as a variable, you have to name it again in order to pass it to the map method:
```
[the remark this answers is not in the document]
Comment (rwieruch) [reply javascript-map.md]: Fixed https://github.com/rwieruch/blog_robinwieruch_content/pull/83/commits/5c3df4bb70b4bf9d0725b1ce67d53f69f3d3f37c :)

**Comment by rwieruch** [comment]
Things done for the review (see https://github.com/rwieruch/blog_robinwieruch_content/pull/83/commits/051f5244aefc0e0aa278f74da99f73bb1e72077c):

**Reading Improvements:**

- [x] variable naming being more elaborate rather than shortened
- [x] removing a few extra lines between paragraphs to combine paragraphs
- [x] headline wording refinements
- [x] mentioning anonymous function in the beginning of the article
- [x] removing several inline code blocks for improved readability + SEO (e.g. `map` to Array Map Method, Map Method)
- [x] simplified example for third argument (array) of the callback function, please check again @benjamminj 

**Syntax:**

- [x] links that point outside of the website's domain are using `{{% a_blank "my article name" "url" %}}` instead of `[my article name](url)`
- [x] usage of `*` instead of `_` for keeping the syntax across articles consistent
- [x]

**Todo: **

- [ ] Link to Kyle Simpson's Article for Prototype Chain

**Considerations:**

- [ ] let's write our own Arrow Function Deep Dive article for linking it in this article later
- [ ] link map, reduce, filter articles later to each other

**Discussion:**

I am using this Syntax as well, but I have heard from people reading my tutorials that it isn't very beginner friendly:

```
myArray.map((value, index, arr) =>
  arr[index - 1] ? value + arr[index - 1] : value,
);
```

Maybe there is a Prettier configuration to get rid of this function's body inlining with a comma. Not sure though.

**Comment by benjamminj** [comment]
Is this the Kyle Simpson article you were thinking of? From YDKJS
https://github.com/getify/You-Dont-Know-JS/blob/master/this%20%26%20object%20prototypes/ch5.md

**Comment by benjamminj** [comment]
RE: the Prettier config I believe the option is `trailingComma: "es5"`, right? Since the comma is _technically_ after the arrow function. Setting to `es5` only sets trailing commas after objects & arrays.

https://prettier.io/docs/en/options.html#trailing-commas

**Comment by rwieruch** [comment]
- [x] added the Kyle Simpson article
- [ ] try Prettier Configuration suggested by @benjamminj

**Comment by rwieruch** [comment]
@benjamminj I merge the PR for now. If you feel anything needs to be changed, just open a new one :)

**Pull request merged**